scheduler/src/cook/mesos/sandbox.clj

```diff
@@ -212,63 +178,23 @@
-                (send pending-sync-agent clear-pending-sync-hostname hostname :sync)
-                (sync-agent-sandboxes publisher-state framework-id hostname)
-                (recur remaining-hostnames true))
-
-              :else
-              (recur remaining-hostnames pending-sync-agent-send-performed))))))
-    {:error-handler (fn sandbox-publisher-error-handler [ex]
-                      (log/error ex "Sync of sandbox directories on pending hosts failed"))}))
-
 (defn prepare-sandbox-publisher
   "This function initializes the sandbox publisher as well as helper functions to send individual
    sandbox entries and trigger sandbox syncing of all tasks on a mesos agent.
    It returns a map with the following entries:
+   :hostname->task-id->sandbox-directory-fn - fn that returns the sandbox directory given the hostname and task-id.
    :mesos-agent-query-cache - the cache that throttles the state sync calls to the mesos agents.
    :publisher-cancel-fn - fn that take no arguments and that terminates the publisher.
    :task-id->sandbox-agent - The agent that manages the task-id->sandbox aggregation and publishing."
-  [framework-id datomic-conn publish-batch-size publish-interval-ms sync-interval-ms max-consecutive-sync-failure
-   mesos-agent-query-cache]
-  (let [pending-sync-agent (agent {:framework-id framework-id
-                                   :host->consecutive-failures {}
-                                   :pending-sync-hosts #{}}) ;; stores the names of hosts pending sync
-        task-id->sandbox-agent (agent {}) ;; stores all the pending task-id->sandbox state
-        publisher-state {:datomic-conn datomic-conn
-                         :mesos-agent-query-cache mesos-agent-query-cache
-                         :pending-sync-agent pending-sync-agent
+  [framework-id datomic-conn publish-batch-size publish-interval-ms mesos-agent-query-cache]
+  (let [task-id->sandbox-agent (agent {}) ;; stores all the pending task-id->sandbox state
+        publisher-state {:mesos-agent-query-cache mesos-agent-query-cache
                          :task-id->sandbox-agent task-id->sandbox-agent}
         publisher-cancel-fn (start-sandbox-publisher
                               task-id->sandbox-agent datomic-conn publish-batch-size publish-interval-ms)
-        syncer-cancel-fn (start-host-sandbox-syncer publisher-state sync-interval-ms max-consecutive-sync-failure)]
+        hostname->task-id->sandbox-directory-fn (fn hostname->task-id->sandbox-directory-fn [agent-hostname task-id]
+                                                  (hostname->task-id->sandbox-directory
+                                                    publisher-state framework-id agent-hostname task-id))]
     (assoc publisher-state
-      :publisher-cancel-fn publisher-cancel-fn
-      :syncer-cancel-fn syncer-cancel-fn)))
+      :hostname->task-id->sandbox-directory-fn hostname->task-id->sandbox-directory-fn
```

Comment:
I really don't like the name of this function, is there something shorter that makes sense?

Reply:
Renamed to `retrieve-sandbox-directory-from-agent`